### DISCUSSION TITLE
Keyboard does not work anymore

### DISCUSSION THREAD

**Discussion by ghost** [rern/rAudio#798] [General]
In the past I have used a wireless keyboard with a USB-dongle to select next/previous/pause tracks on rAudio-1.
When playing random files from the library ("dice"-symbol), I can skip songs I don't want for that moment.
This function does not work anymore. 
The USB-dongle connected to the Raspberry Pi connects as a HID device I think.
I've also tested a wired keyboard, no action also.
I'm using a Raspberry 4 (latest software and updates) with a Hifiberry Digi+ Pro (Great musicplayer, Thank you Rern!).

**Comment by ghost** [comment]
Please can somebody connect a keyboard and check if this "next, previous, pause" functionality is working at their player? 
This to see if I am doing something wrong, or this option really has stopped working. Thank you in advance for helping.

**Reply by apastuszak** [reply]
Which keyboard shortcuts are you using?

**Comment by ghost** [comment]
@apastuszak, first of all, thank you for trying to help.
I'm using the 'standard' keys: left/right for previous/next track, spacebar to pause track. Please see the picture as it is in the player.
Up and down for changing the volume will not work, as the Hifiberry Digi+ Pro just sends through the bitstream unaltered.
Maybe this can be a clue?

![afbeelding](https://user-images.githubusercontent.com/96625068/154065577-dfe42ea8-7588-47cb-8757-57c2ea95ddd4.png)

**Comment by rern** [comment]
Just tested and no issues found except enter key firing twice which will be fixed by next update.

**Reply by ghost** [reply]
Hi Rern, thank you for investigating the keyboard problem. 

No issues found on your side, but over here still problems :-)
I'm not sure if we do have the same test setup. I have a Raspberry 4 with 7" display and Digi+ pro.

If I start a browser on my laptop and connect to raudio.local, 
everything is working fine (arrow keys on the laptop do their job like pause, next, previous). You are correct. No issues there.

If I connect a (wired) USB-keyboard directly to the Raspberry: nothing happens when hitting the keys over there. 
That is where the problem is: at the local keyboard.

I have also setup another Raspberry 4, bare, no screen and no Digi+ connected. Fresh software installed. 
Still no response from the (wired) keyboard. This function worked fine for me in the past.
Is it possible that the software sees the keyboard as a USB drive, and not as a HID device?

Sorry for asking again, but can you give it another try? Thank you in advance.

**Comment by ghost** [comment]
I've been testing a little more right now. Some keys do function correctly!
F1 opens the chrome help. F11 shows/closes full screen mode. 

I'm typing on a standard US querty keyboard.
Is the keyboard-layout causing the problem?
